### ISSUE TITLE
Can We Speed Up Batch Inference?

### ISSUE THREAD

**Issue by yangfy2023** [2U1/Qwen2-VL-Finetune#137]
I attempted to perform inference on my test dataset and found the process to be quite slow when dealing with large data volumes. To improve efficiency, I tried batch inference using model.generate(). While this approach offered some performance gains, I noticed that the time cost still increases approximately linearly with the batch size.

I'm wondering if there are more efficient methods or frameworks available to accelerate inference. For example, would using a library like vLLM help improve the throughput and reduce latency?

**Comment by Darsha-1** [comment]
@yangfy2023 @2U1 Hi, How are you doing batch inference? I have a test set which is similar to training set. I want to produce outputs same as in training outputs for these too. Thanks
